redbeams-api/src/main/java/com/sequenceiq/redbeams/api/endpoint/v4/databaseserver/DatabaseServerV4Endpoint.java

```diff
@@ -77,6 +77,18 @@ DatabaseServerV4Response getByName(
             @ApiParam(DatabaseServerParamDescriptions.NAME) @PathParam("name") String name
     );
 
+    @GET
+    @Path("clusterCrn/{clusterCrn}")
+    @ApiOperation(value = DatabaseServerOpDescription.GET_BY_CLUSTER_CRN,
+            notes = DatabaseServerNotes.GET_BY_CLUSTER_CRN,
+            nickname = "getDatabaseServerByClusterCrn")
+    DatabaseServerV4Response getByClusterCrn(
+            @ValidCrn(resource = CrnResourceDescriptor.ENVIRONMENT) @NotNull @ApiParam(value = DatabaseServerParamDescriptions.ENVIRONMENT_CRN, required = true)
+            @QueryParam("environmentCrn") String environmentCrn,
+            @ValidCrn(resource = { CrnResourceDescriptor.DATALAKE, CrnResourceDescriptor.DATAHUB })
+            @NotNull @ApiParam(value = DatabaseServerParamDescriptions.NAME, required = true) @PathParam("clusterCrn") String clusterCrn
```

Comment:
Note: I think this will have to be updated soon to permit any non-environment CRNs for sake of looking up DB servers for non-CB "clusters" (e.g. DFX). See [this](https://github.com/hortonworks/cloudbreak/pull/10459/files#diff-11d87b3ace4084117083bb0317790b6c0f709d00603e662c2a6d9581247ce5c0) change in #10459.

CC @gergopapi2

Reply:
but does it block the merge?

Reply:
Certainly not, I just flagged it as something to be fixed once both PRs got merged. 😄

Reply:
as the API can't be modified later I would say it would be better to call this resourceCrn or similar which can be meaningful for other services. But I will leave it to @gergopapi2

Reply:
resourceCrn would be nicer yes because other services also want to use redbeams

Reply:
I do agree, however the name clusterCrn is already out in the AllocateDatabaseServerV4Request. For consistency it would make sense to keep the name I think.

Reply:
leaved as `clusterCrn` but changed the `@ValidCrn` as per @lajosrodek  suggested